src/Uno.UI.Composition/Composition/ShapeVisual.skia.cs

```diff
@@ -31,6 +34,19 @@ internal override void Paint(in PaintingSession session)
 		base.Paint(in session);
 	}
 
+	private protected override void OnPropertyChangedCore(string? propertyName, bool isSubPropertyChange)
+	{
+		base.OnPropertyChangedCore(propertyName, isSubPropertyChange);
+		if (propertyName == nameof(Shapes))
+		{
+			_needsContinuousUpdates = _shapes?.OfType<CompositionSpriteShape>().Any(s => s.FillBrush is CompositionEffectBrush { HasBackdropBrushInput: true } or IOnlineBrush { IsOnline: true }) ?? false;
```

Comment:
We get notifications to changes on the collection for free because of the `CompositionObject._contextEntries`. I manually tested that this indeed propagates since I had the same concerns.